pkg/auth/oauth/handlers/interfaces.go

```diff
@@ -25,6 +26,50 @@ type AuthenticationRedirector interface {
 	AuthenticationRedirect(w http.ResponseWriter, req *http.Request) (err error)
 }
 
+// AuthenticationRedirectors is a collection that provides a list of
+// acceptable authentication mechanisms
+type AuthenticationRedirectors struct {
+	Count         int
+	Names         []string
+	RedirectorMap map[string]AuthenticationRedirector
+}
+
+// Add a name and a matching redirection routine to the list
+// The entries added here will be retained in FIFO ordering.
+func (ar *AuthenticationRedirectors) Add(name string, redirector AuthenticationRedirector) {
+	// Initialize the map the first time
+	if ar.RedirectorMap == nil {
+		ar.RedirectorMap = make(map[string]AuthenticationRedirector, 1)
+	}
+
+	// Be overprotective and check whether this name is already in use
```

Comment:
unnecessary, either overwrite or ignore on duplicates... I'm fine with either as long as it's documented